### PR TITLE
Update export catalog

### LINKED ISSUE #886: Implement `DataCatalog` export with new class model.
### Kind of request

Changing existing functionality

### Enhancement Description

With the new classes, the `DataCatalog`s export functionality needs to be implemented.

### Use case

Export HydroMT data.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Jaapel**
## Issue addressed
Fixes #886 

## Explanation
I moved the `DataCatalog.export_data` to the new class structure.
Next to using the correct classes, I had to:
- Fixed a bug in the migrate_datacatalog script to make sure that the `tileindex` `option` ended up in the resolver
- Changed the exported uri to something sensible if the format changes during export. This behaviour is included in the driver.
- Changed some catalog references to match the 0.9 and 1.0 references. I wonder how that ever worked in v0.

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [ ] Updated documentation
- [ ] Updated changelog.rst

## Data/Catalog checklist
- [ ] `data/catalogs/predefined_catalogs.yml` has not been modified.
- [ ] None of the old `data_catalog.yml` files have been changed
- [ ] `data/chagnelog.rst` has been updated
- [ ] new file uses `LF` line endings (done automatically if you used `update_versions.py`)
- [ ] New file has been tested locally
- [ ] Tests have been added using the new file in the test suite

### CODE REVIEW

**Comment by savente93** [comment]
I'll give it an early review tomorrow

File: examples/data/geodataset_catalog.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
+meta:
+  root: examples/data
```
Comment (savente93) [inline examples/data/geodataset_catalog.yml:3]: so this is the old way of registering a root, now the format should be: 
```yml
meta:
    roots:
        - examples/data
```

I'm not sure we ever discussed whether we want to deprecate the old format or not. What do you think?

**Review by savente93 — COMMENTED** [review]
I think this is the best implementation we have time/resources for now. Aside from the one comment about keeping the old root format in the yaml I think this is okay. Once we resolve that I'll be happy to approve

File: examples/data/geodataset_catalog.yml (hunk 1 above)
Comment (Jaapel) [reply examples/data/geodataset_catalog.yml:3]: These are still in the current data catalogs. We can decide to deprecate in another PR

**Review by savente93 — APPROVED** [review]
lets go

**Pull request merged**